lib/cds_importer/entity_mapper/base_mapper.rb

```diff
@@ -11,6 +12,12 @@ class << self
         attr_accessor :entity_class, :entity_mapping, :mapping_path, :mapping_root, :exclude_mapping,
                       :entity_mapping_key_as_array, :entity_mapping_keys_to_parse
 
+        def before_oplog_inserts(&block)
```

Comment:
Reading and writing via the same method feels confusing because they are used in different contexts - the write use case is intended for within the class whilst the read use case is external where there is less contextual information

Could this instead be

`def before_oplog_inserts` for creating the callback, this could be `private`?

`def before_oplog_insert_callbacks` for retrieving them - this would then be clearer at the callsite (ie EntityMapper)

Reply:
We could separate these, yeah